lib/std/static_string_map.zig

```diff
@@ -0,0 +1,502 @@
+            allocator.free(self.len_indexes[0..self.len_indexes_len]);
+            allocator.free(self.kvs.keys[0..self.kvs.len]);
+            allocator.free(self.kvs.values[0..self.kvs.len]);
+            allocator.destroy(self.kvs);
+        }
+
+        const SortContext = struct {
+            keys: [][]const u8,
+            vals: []V,
+
+            pub fn lessThan(ctx: @This(), a: usize, b: usize) bool {
+                return ctx.keys[a].len < ctx.keys[b].len;
+            }
+
+            pub fn swap(ctx: @This(), a: usize, b: usize) void {
+                std.mem.swap([]const u8, &ctx.keys[a], &ctx.keys[b]);
+                std.mem.swap(V, &ctx.vals[a], &ctx.vals[b]);
+            }
+        };
+
+        fn initSortedKVs(
+            self: *Self,
+            kvs_list: anytype,
+            sorted_keys: [][]const u8,
+            sorted_vals: []V,
+        ) void {
+            for (kvs_list, 0..) |kv, i| {
+                sorted_keys[i] = kv.@"0";
+                sorted_vals[i] = if (V == void) {} else kv.@"1";
+                self.min_len = @intCast(@min(self.min_len, kv.@"0".len));
+                self.max_len = @intCast(@max(self.max_len, kv.@"0".len));
+            }
+            mem.sortUnstableContext(0, sorted_keys.len, SortContext{
+                .keys = sorted_keys,
+                .vals = sorted_vals,
+            });
+        }
+
+        fn initLenIndexes(self: Self, len_indexes: []u32) void {
+            var len: usize = 0;
+            var i: u32 = 0;
+            while (len <= self.max_len) : (len += 1) {
+                // find the first keyword len == len
+                while (len > self.kvs.keys[i].len) {
+                    i += 1;
+                }
+                len_indexes[len] = i;
+            }
+        }
+
+        /// Checks if the map has a value for the key.
+        pub fn has(self: Self, str: []const u8) bool {
+            return self.get(str) != null;
+        }
+
+        /// Returns the value for the key if any, else null.
+        pub fn get(self: Self, str: []const u8) ?V {
+            if (self.kvs.len == 0)
+                return null;
+
+            return self.kvs.values[self.getIndex(str) orelse return null];
+        }
+
+        pub fn getIndex(self: Self, str: []const u8) ?usize {
+            const kvs = self.kvs.*;
+            if (kvs.len == 0)
+                return null;
+
+            if (str.len < self.min_len or str.len > self.max_len)
+                return null;
+
+            var i = self.len_indexes[str.len];
+            while (true) {
+                const key = kvs.keys[i];
+                if (key.len != str.len)
+                    return null;
+                if (eql(key, str))
+                    return i;
+                i += 1;
+                if (i >= kvs.len)
+                    return null;
+            }
+        }
+
+        /// Returns the longest key, value pair where key is a prefix of `str`
+        /// else null.
```

Comment:
Nit: This makes it sound as if the value contributes to the length calculation.
```suggestion
        /// Returns the key-value pair where key is the longest prefix of `str`
        /// else null.
```

Reply:
Thanks. :+1: . I'll make sure to change this next review if `getLongestPrefix()` is approved.

Reply:
Ah merged already.  Will have to be a follow up.

Reply:
I've added these changes in #19823 along with a note about time complexity